src/ovp_pipeline/_truth_helpers.py

```
@@ -0,0 +1,451 @@
+    requested_pack = candidates[0]
+    db = _db_path(vault_dir)
+    row = None
+    try:
+        with sqlite3.connect(db) as conn:
+            row = conn.execute(
+                "SELECT 1 FROM truth_projections WHERE pack = ? LIMIT 1",
+                (requested_pack,),
+            ).fetchone()
+            if row is None:
+                row = conn.execute(
+                    f"SELECT 1 FROM {table_name} WHERE pack = ? LIMIT 1",
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)